ParkenDD/Colors.swift

```diff
@@ -22,20 +22,34 @@ struct Colors {
 	*/
 	static func colorBasedOnPercentage(_ percentage: Double, emptyLots: Int) -> UIColor {
 
-		let hue = 1 - (percentage * 0.3 + 0.7) // I want to limit this to the colors between 0 and 0.3
+        let lotDensityColor: UIColor = {
+            switch percentage {
+            case 0.6...1:
+                return UIColor(rgba: "#006a39")
+            case 0.16..<0.6:
+                return UIColor(rgba: "#1daa8c")
+            case 0.01..<0.16:
+                return UIColor(rgba: "#7f0304")
+            default:
+                return UIColor(rgba: "#5c5c5c")
+            }
+        }()
 
 		let useGrayscale = UserDefaults.standard.bool(forKey: Defaults.grayscaleUI)
 		if useGrayscale {
 			if emptyLots <= 0 {
 				return UIColor(red: 0.8, green: 0.8, blue: 0.8, alpha: 1.0)
 			}
-			return UIColor(red: CGFloat(1 - (hue+0.2)), green: CGFloat(1 - (hue+0.2)), blue: CGFloat(1 - (hue+0.2)), alpha: 1.0)
+            return lotDensityColor
 		}
 
 		if emptyLots <= 0 {
-			return UIColor(hue: CGFloat(hue), saturation: 0.54, brightness: 0.7, alpha: 1.0)
+            return UIColor(hue: lotDensityColor.hsbaComponents.hue,
```

Comment:
Explain why you're subtracting `-0.1`

Reply:
In the original code, the only difference between the returned hue if the if-statement is true is that the brightness was 0.1 lower than the hue returned resulting from a false if-statement.  In other words, if the contents of the if-statement are executed, it returns a hue.  If they are not executed it returns the same hue, only 0.1 brighter.  Therefore, the reason I am subtracting 0.1 is to keep the difference in brightness consistent with the existing code.

Reply:
Hehe, I meant, explain it in the code 😄

Reply:
Created a comment. I also saw that I updated the brightness incorrectly and changed that.